src/sentry/incidents/grouptype.py

```diff
@@ -36,7 +36,9 @@ class MetricIssueEvidenceData(EvidenceData[float]):
     alert_id: int
 
 
-class MetricIssueDetectorHandler(StatefulDetectorHandler[QuerySubscriptionUpdate, int]):
+class MetricIssueDetectorHandler(
+    StatefulDetectorHandler[QuerySubscriptionUpdate | AnomalyDetectionUpdate, int | dict]
```

Comment:
For readability, it might be nice to give these types a name:
```suggestion
MetricUpdate = QuerySubscriptionUpdate | AnomalyDetetionUpdate
MetricResult = int | dict


class MetricIssueDetectorHandler(
    StatefulDetectorHandler[MetricUpdate, MetricResult]
```

Then in methods like `extract_value` it gets easier to grok:
```py
def extract_value(self, data_packet: DataPacket[MetricUpdate]) -> MetricResult:
```